prov/util/src/util_mem_hooks.c

```diff
@@ -373,6 +381,85 @@ static bool ofi_is_function_patched(struct ofi_intercept *intercept)
         ((*(uint32_t *) (addr + 16)) & br_mask) == br(0)
 	);
 }
+#elif (defined(__riscv) && defined(__riscv_xlen) && (__riscv_xlen == 64))
+
+/* Registers numbers to use with the move immediate to register.
+ * The destination register is X31 (highest temporary).
+ * Register X28-X30 are used for block shifting and masking.
+ * Register X0 is always zero
+ */
+#define X31 31
+#define X30 30
+#define X0  0
+
+/**
+ * @brief JALR - Add 12 bit immediate to source register, save to destination register, jump and link from destination register
```

Comment:
line lengths are too long, shorten to about 80ish characters

Reply:
if the the alignment attribute is on a new line, would you want it indented under the `unsigned char` or the `patch_data`?